src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue29420.cs

```diff
@@ -0,0 +1,32 @@
+﻿#if TEST_FAILS_ON_IOS && TEST_FAILS_ON_CATALYST // CarouselView Fails to Keep Last Item in View on iOS, android and macOS https://github.com/dotnet/maui/issues/18029, https://github.com/dotnet/maui/issues/29415
+using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+public class Issue29420 : _IssuesUITest
+{
+	public override string Issue => "KeepLastInView Not Working as Expected in CarouselView";
+
+	public Issue29420(TestDevice device) : base(device)
+	{ }
+
+	[Test]
+	[Category(UITestCategories.CarouselView)]
+	public void VerifyCarouselViewKeepLastInViewOnItemInsert()
```

Comment:
Pending snapshots on Android and Windows. Already available in the latest build.
<img width="347" alt="image" src="https://github.com/user-attachments/assets/7dadb7ad-5ff0-47c3-b381-87d21c014ddb" />

Could you commit the images?